qemu/tests/smt_test.py

```diff
@@ -23,20 +23,32 @@ def run(test, params, env):
     :params params: Dictionary with the test parameters.
     :params env: Dictionary with test environment.
     """
+    def run_guest_cmd(cmd, retry=False):
+        """
+        Run cmd inside guest
+        """
+        output = session.cmd_output_safe(cmd)
+        if retry and not output:
+            output = session.cmd_output_safe(cmd)
+        return output
+
     def get_guest_threads():
         """
         Get guest threads number
         """
         if os_type == "linux":
             cmd = params["get_threads_cmd"]
-            output = session.cmd_output_safe(cmd)
+            output = run_guest_cmd(cmd)
             threads = int(re.findall(r":\s*(\d+)", output)[0])
```

Comment:
I wonder if we need update `threads` and `sockets` in L52 to avoid the possible index error too.
Replace index error with specific error message, how do you think ?

Reply:
How about check if output is empty before L33 directly? Then we may not need to check every time after run_guest_cmd.

Reply:
Consider the main reason of index error is the output is None, i think it is a good idea.  
The situation that can not match the expect string with a non-empty output can be ignored i think.

Reply:
Updated, pls help review again.